ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Util/PsiUtil.fs

```diff
@@ -409,6 +409,19 @@ let rec skipIntermediateParentsOfSameType<'T when 'T :> ITreeNode and 'T: not st
 let rec skipIntermediatePatParents (fsPattern: IFSharpPattern) =
     skipIntermediateParentsOfSameType<IFSharpPattern> fsPattern
 
+let rec isPartOfDotLambda (expr: IFSharpExpression) =
+    let exprSkipParens = expr.IgnoreParentParens()
+    // _.M( <caret> )
+    if isNotNull (PrefixAppExprNavigator.GetByArgumentExpression(exprSkipParens)) &&
+       not (expr :? IParenExpr) &&
+       not (expr :? IUnitExpr) then false
+    // _.[ <caret> ]
```

Comment:
Are there any tree tests covering this kind of cases?

Reply:
We have a testcase for `_.ToString()`. 
`isPartOfDotLambda` is more about the semantics of what is a direct part of a dot lambda. There are tests for this in `IntroduceVar`.

Reply:
What would be nice is this: if we have some logic that depends on a particular tree structure it would be really helpful to be able to check the tree in the test baselines. In that way, the brackets tests would be a good addition here.